notebooks/HASP/COSLifetimePositions/COSLifetimePositions.ipynb

```diff
@@ -0,0 +1,965 @@
+    "        target_name='WD*0308-565',\n",
+    "        dataproduct_type=\"SPECTRUM\"\n",
+    "    )\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "aa19d244",
+   "metadata": {},
+   "source": [
+    "This query returns all available data, but we want only the x1d files of spectra taken with the G130M grating.  (This takes a few minutes, so be patient.)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b390c561",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "Observations.download_products(\n",
+    "    wg7_products,\n",
+    "    download_dir=str(cos_data_dir),\n",
+    "    filters='G130M',\n",
+    "    productSubGroupDescription=[\"X1D\"]\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "902f9ce3",
+   "metadata": {},
+   "source": [
+    "When we download data using astroquery, it creates a directory called mastDownload/HST. Each dataset goes into a separate subfolder within that directory.  Before running the combination script, we must move all of the x1d files into a single directory. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d6e26ca4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "try:\n",
+    "    # The path to all obs_id folders\n",
+    "    mast_path = f\"{cos_data_dir}/mastDownload/HST/\"\n",
+    "\n",
+    "    # Check if mastDownload exists\n",
+    "    if not os.path.exists(mast_path):\n",
+    "        print(f\"Directory {mast_path} doesn't exist.\")\n",
+    "\n",
+    "    # Getting a list of all obs_id folders. Each folder contains the FITS files\n",
+    "    obs_id_dirs = os.listdir(mast_path)\n",
+    "\n",
+    "    # Iterating through sub-folders to change the path of each FITS file\n",
+    "    for obs_id in obs_id_dirs:\n",
+    "        # This is the path to each obs_id folder\n",
+    "        obs_id_path = os.path.join(mast_path, obs_id)\n",
+    "\n",
+    "        # Getting list of FITS files in /mastDownload/HST/<obs_id> folder\n",
+    "        cos_files = glob.glob(obs_id_path + \"/*fits\")\n",
+    "\n",
+    "        # Iterating through each of these files to change their path individually\n",
+    "        # We will be moving them to /cos_data\n",
+    "        for file in cos_files:\n",
+    "            file_path = Path(file)\n",
+    "            new_path = cos_data_dir / file_path.name\n",
+    "            shutil.move(file, new_path)\n",
+    "\n",
+    "    # Now we can remove the mastDownload directory\n",
+    "    if os.path.exists(mast_path):\n",
+    "        shutil.rmtree(f\"{cos_data_dir}/mastDownload/\")\n",
+    "\n",
+    "except Exception as e:\n",
+    "    print(f\"An error occurred: {e}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ce316550",
+   "metadata": {},
+   "source": [
+    "We want only the files taken at lifetime positions 1 and 4 and with CENWAVE > 1250, so delete everything else."
```

Comment:
is there a way to filter this beforehand in the astroquery query, rather than deleting the files? are these are columns in the data base?